src/main/java/bio/terra/service/tabulardata/google/BigQueryPdao.java

```diff
@@ -1982,6 +1973,35 @@ public void validateDeleteRequest(
     return values;
   }
 
+  private static final String snapshotDataTemplate =
+      "SELECT * FROM `<project>.<snapshot>.<table>` LIMIT <count>";
+
+  public List<Map<String, Object>> getSnapshotTable(
+      Snapshot snapshot, String tableName, Integer count) throws InterruptedException {
+    final BigQueryProject bigQueryProject = BigQueryProject.from(snapshot);
+    final String sql =
+        new ST(snapshotDataTemplate)
+            .add("project", bigQueryProject.getProjectId())
+            .add("snapshot", snapshot.getName())
+            .add("table", tableName)
+            .add("count", count)
+            .render();
+    final TableResult result = bigQueryProject.query(sql);
+
+    return aggregateTableColumnsRows(result);
```

Comment:
Why not call `getSnapshotTableData()` here? (`BigQueryProject.from()` is called twice that way, but its result is cached so there shouldn't be any performance penalty)
```suggestion
    return getSnapshotTableData(snapshot, sql);
```